src/dataflow/components/dataflow-program.tsx

```diff
@@ -178,7 +178,9 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     this.programEditor.clear();
   }
   private deleteSelectedNodes = () => {
-    this.programEditor.selected.list.forEach((n: Node) => {
+    const selectedNodes = this.programEditor.selected.list;
```

Comment:
This may just be a reference. Adding `.slice()` will give you a copy if it's a standard JavaScript array.

Reply:
Updated